### ISSUE TITLE
W2V-Unsupervised: sentence utterance from uppermost features, clustering based on n-th block

### ISSUE THREAD

**Issue by prokotg** [pytorch/fairseq#3785]
#### What is your question?

`fairseq/examples/wav2vec/unsupervised/scripts/wav2vec_apply_cluster_faiss.py` first calculates top-most features from W2V2.0 via `wav2vec_extract_features.py`:

https://github.com/pytorch/fairseq/blob/741fd138c69eebc1fac09b590047da85cdeafe2d/examples/wav2vec/unsupervised/scripts/wav2vec_extract_features.py#L55-L66

and these features are later on starting with PCA:

https://github.com/pytorch/fairseq/blob/741fd138c69eebc1fac09b590047da85cdeafe2d/examples/wav2vec/unsupervised/scripts/prepare_audio.sh#L68

but clustering scripts use features from `n-th` block:

https://github.com/pytorch/fairseq/blob/741fd138c69eebc1fac09b590047da85cdeafe2d/examples/wav2vec/unsupervised/scripts/wav2vec_cluster_faiss.py#L100-L107

**Is this ok?** It is not clear for me if that's how it is suppose to be, even after re-reading the paper which says:

> Our goal is to learn a model which can map from audio representations clt to phonemes using no supervision. However, the representations of the uppermost block of wav2vec 2.0 may not be well suited for this task.

And later on: 

>  Once the speech audio representations are segmented, we compute a 512-dimensional PCA over all speech representations output by wav2vec 2.0 for the training set.

What exactly is `all speech representations output by wav2vec 2.0 ` here?

#### What's your environment?

 - fairseq Version (e.g., 1.0 or master): master

**Comment by prokotg** [comment]
Ok, I can see now that these are taken from the same layer. Then why do we have to calculate features twice? (there are some subtle differences like in one script there is task's normalization taken into account )
